lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -166,12 +166,18 @@ WasmBinaryReader::ProcessSection(SectionCode sectionId, bool isEntry /*= true*/)
         break; // This section is not used by bytecode generator, stay in decoder
     }
     case bSectDataSegments:
-        // TODO: Populate Data entry info
-        ReadConst<UINT32>();  // dest addr in module memory
-        ReadConst<UINT32>();  // source offset (?)
-        ReadConst<UINT32>();  // size
-        ReadConst<UINT8>();  // init
-        m_moduleState.count++;
+        m_moduleState.size = LEB128(length);
+        if (m_moduleState.size == 0)
+        {
+            // TODO: Probably can factor this out and validate all count fields.
+            ThrowDecodingError(L"Illegal data segment entry count");
```

Comment:
Since you'll have to rebase on top of my integrate merge, might as well fix this to `_u("Illegal data segment entry count")`